haskell/test/Test/Ligo/BaseDAO/Proposal.hs

```diff
@@ -799,15 +839,31 @@ dropProposal originateFn = do
 
   key3 <- createSampleProposal 3 owner1 dao
 
-  withSender (AddressResolved admin) $ do
+  -- -- TODO: Change to `guardian`, but currently it is not working may be due to sender <> source
+  withSender (AddressResolved owner1) $ do
     call dao (Call @"Drop_proposal") key1
```

Comment:
@pasqu4le I have added `proposal_flush_time`, and `proposal_expired_time`, update `flush` and `drop_proposal` entrypoints, and also update the tests.

One thing that I couldn't figure out is writing the test for `drop_proposal` such that the `guardian` can drop any proposals. 
- Currently, I set `guardian = admin` in the tests, and I think the check `sender <> source` cause this test to fail.
- I don't fully understand how `SOURCE` works, hope to get your advice on what the proper value should `guadian` be.

Reply:
@rinn7e AFAIU you should set guardian address to the address of a contract that can call the `drop_proposal` entrypoint, and then start by calling the guardian. So the SOURCE would be the address that you used to call the guardian, but SENDER will be the address of the guardian itself.

Reply:
@sras is correct, for tests all we need is a very simple `guardian` contract that has no checks and:
- stores the DAO address
- its whole parameter is the same as `drop_proposal`
- when called it will "pass the call forward", transferring to the DAO's `drop_proposal`

Reply:
That makes sense. Turn out that, the current `OriginateFn m` which uses the tuple, is a bit troublesome to add a new field `guardian` to it. So I update it to be a record instead and this is quite a big change (mostly renaming), and I put it into a new commit.

The `guardian` contract is done, and I've added to the tests, hope to get your reviews on this.